vault/generate_root_recovery.go

```diff
@@ -0,0 +1,31 @@
+package vault
+
+import (
+	"context"
+
+	"github.com/hashicorp/vault/sdk/helper/base62"
+	"go.uber.org/atomic"
+)
+
+// GenerateRecoveryTokenStrategy is the strategy used to generate a
+// recovery token
+func GenerateRecoveryTokenStrategy(token *atomic.String) GenerateRootStrategy {
+	return &generateRecoveryToken{token: token}
+}
+
+// generateRecoveryToken implements the GenerateRootStrategy and is in
+// charge of creating recovery tokens.
+type generateRecoveryToken struct {
+	token *atomic.String
+}
+
+func (g *generateRecoveryToken) generate(ctx context.Context, c *Core) (string, func(), error) {
+	id, err := base62.Random(TokenLength)
+	if err != nil {
+		return "", nil, err
+	}
+	token := "s." + id
```

Comment:
Since this is not a service token, should we remove the prefix or should we employ a `r.` to indicate recovery operation token?

Reply:
Sure, I think that's a fine idea (r.)

Reply:
Done